src/webots/nodes/WbBackground.cpp

```diff
@@ -311,113 +277,139 @@ void WbBackground::applySkyBoxToWren() {
   QVector<float *> hdrImageData;
   QVector<QImage *> regularImageData;
 
+  // 1. Load the background.
+  mCubeMapTexture = wr_texture_cubemap_new();
   try {
-    QString suffix = "";
+    bool allUrlDefined = true;
+    bool atLeastOneUrlDefined = false;
+
     for (int i = 0; i < 6; ++i) {
-      if (mUrlFields[i]->size() == 0)
-        throw QString();
+      if (mUrlFields[i]->size() == 0) {
+        allUrlDefined = false;
+        textureUrls[i] = "";
+        continue;
+      } else
+        atLeastOneUrlDefined = true;
 
       textureUrls[i] = WbUrl::computePath(this, "textureBaseName", mUrlFields[i]->item(0), false);
-
-      if (textureUrls[i].isEmpty())
-        throw QString();
-
-      QString newSuffix = QFileInfo(textureUrls[i]).suffix();
-      if (i > 0 && newSuffix != suffix)
-        throw tr("Inconsistent image format.");
-
-      suffix = newSuffix;
     }
 
-    if (suffix == "hdr") {
-      wr_texture_set_internal_format(WR_TEXTURE(mCubeMapTexture), WR_TEXTURE_INTERNAL_FORMAT_RGB32F);
-
-      for (int i = 0; i < 6; i++) {
-        int width, height, nrComponents;
-        float *data = stbi_loadf(textureUrls[i].toUtf8().constData(), &width, &height, &nrComponents, 0);
-
-        if (width != height)
-          throw tr("The texture '%1' is not a square image (its width doesn't equal its height).").arg(textureUrls[i]);
-        if (i > 0 && width != edgeLength)
-          throw tr("Texture dimension mismatch between '%1' and '%2'").arg(lastFile).arg(textureUrls[i]);
+    if (!allUrlDefined)
+      throw QString(atLeastOneUrlDefined ? tr("Incomplete cubemap") : "");
 
-        hdrImageData.append(data);
-        edgeLength = width;
+    wr_texture_set_internal_format(WR_TEXTURE(mCubeMapTexture), WR_TEXTURE_INTERNAL_FORMAT_RGBA8);
 
-        wr_texture_cubemap_set_data(mCubeMapTexture, reinterpret_cast<const char *>(data),
-                                    static_cast<WrTextureOrientation>(i));
-      }
-    } else {
-      wr_texture_set_internal_format(WR_TEXTURE(mCubeMapTexture), WR_TEXTURE_INTERNAL_FORMAT_RGBA8);
+    bool alpha = false;
+    for (int i = 0; i < 6; i++) {
+      QImageReader imageReader(textureUrls[i]);
+      QSize textureSize = imageReader.size();
 
-      bool alpha = false;
-      for (int i = 0; i < 6; i++) {
-        QImageReader imageReader(textureUrls[i]);
-        QSize textureSize = imageReader.size();
+      if (textureSize.width() != textureSize.height())
+        throw tr("The texture '%1' is not a square image (its width doesn't equal its height).").arg(imageReader.fileName());
+      if (i > 0 && textureSize.width() != edgeLength)
+        throw tr("Texture dimension mismatch between '%1' and '%2'").arg(lastFile).arg(imageReader.fileName());
```

Comment:
Are those checks really usefull?
I don't see any good reason for keeping this constraint.

Reply:
Yes, they are useful. Without, it looks like this 😉 :

![physically_based_rendering](https://user-images.githubusercontent.com/866788/67684452-81850d00-f993-11e9-8d40-0cbfbb48c5ec.png)

Wren loads these images as a cubemap which offers useful accessor abilities, but also this constraint about dimension.